### PR TITLE
Add Kerr solution to XCTS equations

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Use the existing Kerr solution to compute variables for the XCTS equations.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

File: src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp [hunk 1]
```diff
@@ -544,9 +544,9 @@ class KerrSchild : public MarkAsAnalyticSolution {
     static constexpr double null_vector_0_ = -1.0;
   };
 
-  double mass_{1.0};
-  std::array<double, volume_dim> dimensionless_spin_{{0.0, 0.0, 0.0}};
-  std::array<double, volume_dim> center_{{0.0, 0.0, 0.0}};
+  double mass_{std::numeric_limits<double>::signaling_NaN()};
+  std::array<double, volume_dim> dimensionless_spin_{};
```
Comment (nilsdeppe) [inline src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp]: maybe `make_array<volume_dim>(signaling_NaN())`?

File: src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp [hunk 2]
```diff
@@ -295,7 +295,6 @@ class KerrSchild : public MarkAsAnalyticSolution {
     return dimensionless_spin_;
   }
 
- private:
```
Comment (nilsdeppe) [inline src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp:298]: if it's only needed in derived classes, could/should it be `protected` instead?

File: src/PointwiseFunctions/AnalyticSolutions/Xcts/Kerr.hpp [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+ * The conformal factor in this solution is set to \f$\psi=1\f$, so the
+ * conformal background-metric is the spatial Kerr metric. It is possible to
+ * choose a different \f$\psi\f$ so the solution is non-trivial in this
+ * variable, though that is probably only useful for testing and currently not
+ * implemented. It should be noted, however, that the combination of
+ * \f$\psi=1\f$ and apparent-horizon boundary conditions poses a hard problem to
+ * the nonlinear solver when starting at a flat initial guess. This is because
+ * the strongly-nonlinear boundary-conditions couple the variables in such a way
+ * that the solution is initially corrected away from \f$\psi=1\f$ and is then
+ * unable to recover. A conformal-factor profile such as \f$\psi=1 +
```
Comment (nilsdeppe) [inline src/PointwiseFunctions/AnalyticSolutions/Xcts/Kerr.hpp:179]: I found this bit of extra info on the conformal factor very helpful :)

**Review by nilsdeppe — DISMISSED** [review]
Please feel free to squash any changes immediately :)

File: src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp (hunk 2 above)
Comment (nilsvu) [reply src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp:298]: Unfortunately the "registrars" code currently requires the implementation of the Kerr variables be in a separate class so they can be moved to a cpp file, which means this needs to be public (one of the things that the new factory-creation code #3084 will make a lot simpler).

**Comment by nilsvu** [comment]
Rebased and squashed in the one change, thanks @nilsdeppe!

File: src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp (hunk 2 above)
Comment (nilsdeppe) [reply src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp:298]: okay, sounds good. Thanks :)

**Review by nilsdeppe — DISMISSED** [review]

**Comment by nilsvu** [comment]
Rebased on develop to fix the formatting CI failure. Also added the `Tags::Conformal` prefix to the matter tags to work with #3107.

**Comment by nilsvu** [comment]
@fmahebert Would you be willing to review this one?

**Review by nilsdeppe — APPROVED** [review]

**Comment by fmahebert** [comment]
@nilsleiffischer I'm happy to review this, but I may need a few days to get to it. Feel free to get a different reviewer if that's helpful.

**Review by fmahebert — APPROVED** [review]

**Pull request merged**